language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/EnumClass.scala

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf.codegen.backend.java.inner
+
+import com.daml.ledger.javaapi
+import com.digitalasset.daml.lf.data.Ref.Identifier
+import com.digitalasset.daml.lf.iface
+import com.squareup.javapoet._
+import com.typesafe.scalalogging.StrictLogging
+import javax.lang.model.element.Modifier
+
+import scala.collection.JavaConverters._
+
+private[inner] object EnumClass extends StrictLogging {
+
+  def generate(
+      className: ClassName,
+      identifier: Identifier,
+      enum: iface.Enum,
+  ): TypeSpec = {
+
+    logger.info("Start")
```

Comment:
Please wrap this call in `TrackLineage` as `RecordClass`, `TemplateClass` and `VariantClass` do. This is used to give the user the context in which some logging is happening without having to push information down the stack exclusively for logging.

Reply:
done.